python/GafferSceneUI/SetFilterUI.py

```diff
@@ -94,3 +98,148 @@
 	}
 
 )
+
+##########################################################################
+# NodeGadget drop handler
+##########################################################################
+
+__DropMode = enum.Enum( "__DropMode", [ "None_", "Add", "Remove", "Replace" ] )
+
+__originalDragPointer = None
+
+def __setsPlug( node ) :
+
+	for plug in node.children( Gaffer.Plug ) :
+		if Gaffer.Metadata.value( plug, "ui:scene:acceptsSetNames" ) or Gaffer.Metadata.value( plug, "ui:scene:acceptsSetExpression" ) :
+			return plug
+
+	return None
+
+def __dropMode( nodeGadget, event ) :
+
+	if __setsPlug( nodeGadget.node() ) is None :
+		filter = None
+		if nodeGadget.node()["filter"].getInput() is not None :
+			filter = nodeGadget.node()["filter"].source().node()
+		if filter is None :
+			return __DropMode.Replace
+		elif not isinstance( filter, GafferScene.SetFilter ) :
+			return __DropMode.None_
+
+	if event.modifiers & event.Modifiers.Shift :
+		return __DropMode.Add
+	elif event.modifiers & event.Modifiers.Control :
+		return __DropMode.Remove
+	else :
+		return __DropMode.Replace
+
+def __expressionSets( setExpression ) :
+
+	if (
+		"in" in setExpression or
+		"containing" in setExpression or
+		"|" in setExpression or
+		"&" in setExpression or
+		"(" in setExpression
+	) :
+		return [ f"( {setExpression} )" ]
+
+	return setExpression.split( " " )
+
+def __dragEnter( nodeGadget, event ) :
+
+	if not isinstance( event.data, IECore.StringVectorData ) :
+		return False
```

Comment:
I'm torn between just accepting any old list of strings, and wanting to limit it to sets from the SetEditor, possibly with the additional of "path-like" strings (beginning with /) from elsewhere. I feel like _some_ constraints might help users understand what the operation is. What are your thoughts?

Reply:
Restricting it a bit more does seem like a good idea. I've done that in https://github.com/GafferHQ/gaffer/pull/5786/commits/ecdbab3a990d6082b50440856092d8cf2e7607b3.